### PR TITLE
MVC Controller Response - Wrong ContentType #3245

### PR DESCRIPTION
**Pull request by kichalla**
@rynowak

### CODE REVIEW

File: test/WebSites/FormatterWebSite/Controllers/StreamController.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public Stream MemoryStreamWithContentTypeFromProducesWithMultipleValues()
 
         [HttpGet]
         [Produces("text/plain")]
-        public Stream MemoryStreamOverridesContentTypeWithProduces()
+        public Stream MemoryStreamOverridesProducesContentTypeWithResponseContentType()
         {
             // Produces will set a ContentType on the implicit ObjectResult and
```
Comment (kichalla) [inline test/WebSites/FormatterWebSite/Controllers/StreamController.cs]: will change this comment

File: src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs [hunk 2]
```diff
@@ -159,6 +157,15 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
...
             }
 
+            var responseContentType = formatterContext.HttpContext.Response.ContentType;
+            if (!string.IsNullOrEmpty(responseContentType))
+            {
+                contentTypes.Clear();
+                contentTypes.Add(MediaTypeHeaderValue.Parse(responseContentType));
+            }
+
+            ValidateContentTypes(contentTypes);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs]: This whole block should really be inside of `ExecuteAsync` (where you took this code from). If I'm overriding this method, I want to **use** what was passed in, not ignore it.

File: src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs [hunk 3]
```diff
@@ -159,6 +157,15 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 throw new ArgumentNullException(nameof(formatters));
             }
 
+            var responseContentType = formatterContext.HttpContext.Response.ContentType;
+            if (!string.IsNullOrEmpty(responseContentType))
+            {
+                contentTypes.Clear();
+                contentTypes.Add(MediaTypeHeaderValue.Parse(responseContentType));
+            }
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs]: Why does `Response.ContentType` take precedence over something set on the object result? Let's get agreement on the right priority, and add some comments here to that effect. Let's discuss with @lodejard when he's around.

**Comment by rynowak** [comment]
:watch: - let's discuss briefly with Lou. The code changes are good in general, I'm just not sure the priority is right.

**Comment by kichalla** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
d77655fb7334e4fdde99797b9e2579649aa72aeb

**Pull request closed without merging**